### DISCUSSION TITLE
Integrated Terminal does not start on linux.

### DISCUSSION THREAD

**Discussion by Xachman** [apache/netbeans#3917] [Q&A]
While running netbeans on Ubuntu Focal thru vnc on fluxbox I cannot seem to get the terminal to work.

![image](https://user-images.githubusercontent.com/7697774/161319022-6feecd1a-4e99-47ed-8273-77905f37f5fc.png)

All I get is this gray box. I'm assuming its looking for some terminal program that I could install I just dont know what one.

**Comment by mbien** [comment]
any exceptions in the log? do you have some exotic shell installed on your vnc connected focal fluxbox device?

**Reply by Xachman** [reply]
I don't have anything exotic installed. It has bash like any normal Ubuntu install. What logs would you like me to reference? I will say I do the same thing with IntelliJ community and the terminal on that works fine so I just figured this was looking for something specific that's just not installed.

**Reply by Xachman** [reply]
One more thing too if I click on the terminal enough I can start to see what seems to be file not found. You have to be really quick to see it it's like it's a black box that shows up for a second and then disappears and it says something about file not found.

**Reply by negora** [reply]
When you say *if I click on the terminal enough*, Do you mean on the button with a terminal icon, the one with the label/tooltip *Create new local terminal tab*?

**Comment by Xachman** [comment]
Does anyone know what the terminal here should be using for linux?  Is it a program like the gnome terminal or something?

**Comment by neilcsmith-net** [comment]
@Xachman what architecture are you on?

**Reply by Xachman** [reply]
It's ubuntu-focal running in lxc on top of lxd.  Amd epyc cpus.

**Reply by neilcsmith-net** [reply]
Ah, OK.  Mainly wondered if it was Arm.  Debugging through from here might be useful - https://github.com/apache/netbeans/blob/master/ide/dlight.terminal/src/org/netbeans/modules/dlight/terminal/action/TerminalSupportImpl.java#L115

**Comment by bkoehm** [comment]
An unset `SHELL` environment variable was the problem for me.  Making sure the `SHELL` environment variable was set before launching netbeans was the solution in my case.  For example, `export SHELL=/bin/bash`.
